src/pack/make_pack_descriptor.cpp

```diff
@@ -0,0 +1,110 @@
+//========================================================================================
+// (C) (or copyright) 2020-2025. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include <memory>
+#include <regex>
+#include <set>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include "mesh/meshblock.hpp"
+#include "pack/make_pack_descriptor.hpp"
+
+namespace parthenon {
+namespace impl {
+
+PackDescriptor MakePackDescriptorBase(StateDescriptor *psd,
+                                      const std::vector<std::string> &vars,
+                                      const std::vector<bool> &use_regex,
+                                      const std::vector<MetadataFlag> &flags,
+                                      const std::set<PDOpt> &options) {
+  const std::string cache_label{"normal"};
+  using PDCache = PackDescCache<std::vector<std::string>, std::vector<bool>,
+                                std::vector<MetadataFlag>, std::set<PDOpt>>;
+  auto optional_pd =
+      PDCache::CheckForKey(psd, cache_label, vars, use_regex, flags, options);
+  if (optional_pd) return *optional_pd;
+
+  PARTHENON_REQUIRE(vars.size() == use_regex.size(),
+                    "Vargroup names and use_regex need to be the same size.");
+  auto selector = [&](int vidx, const VarID &id, const Metadata &md) {
+    if (flags.size() > 0) {
```

Comment:
This check is not needed (in combination with the following iterator for loop), is it?